app/ai/voice/agents/breeze_buddy/observers/manager.py

```
@@ -94,28 +101,35 @@ async def _run_checks(self, event_name: str):
+                done, _ = await asyncio.wait(
+                    tasks.keys(), return_when=asyncio.FIRST_COMPLETED
+                )
+                for task in done:
+                    obs = tasks.pop(task)
                     try:
-                        await obs.execute_action()
-                    except Exception as e:
-                        logger.error(f"Observer {obs.name} execute_action failed: {e}")
-                    return
+                        result = task.result()
+                    except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)